### PR TITLE
Send before upload

### PR DESCRIPTION
**Pull request by phong9xNcc**

### CODE REVIEW

File: libs/components/src/lib/components/MessageBox/index.tsx [hunk 1]
```diff
@@ -79,8 +79,13 @@ function MessageBox(props: MessageBoxProps): ReactElement {
 		[currentChannelId],
 	);
 
-	const removeAttachmentByUrl = (urlToRemove: string) => {
-		dispatch(referencesActions.removeAttachment({ channelId: currentChannelId || '', urlAttachment: urlToRemove }));
+	const removeAttachmentByUrl = (indexItem: number) => {
```
Comment (nguyentrannhan) [inline libs/components/src/lib/components/MessageBox/index.tsx]: sao co 2 ham removeAttachmentByUrl?

File: libs/components/src/lib/components/MessageBox/FileSelectionButton.tsx [hunk 2]
```diff
@@ -10,41 +15,64 @@ export type FileSelectionButtonProps = {
...
+						results,
+						undefined,
+						true,
+					);
+				})
+				.catch((error) => {
+					console.error('Error uploading files:', error);
+				});
 		}
+		dispatch(referencesActions.removeAttachmentAfterUpload([]));
```
Comment (nguyentrannhan) [inline libs/components/src/lib/components/MessageBox/FileSelectionButton.tsx]: check argument

File: libs/store/src/lib/messages/messages.slice.ts [hunk 3]
```diff
@@ -225,13 +226,11 @@ export const fetchMessages = createAsyncThunk(
 		});
 
 		thunkAPI.dispatch(reactionActions.updateBulkMessageReactions({ messages }));
-		
-		const lastLoadMessageId = messages[messages.length - 1].id
-		const hasMore = firstMessage?.id === lastLoadMessageId
+
+		const lastLoadMessageId = messages[messages.length - 1].id;
+		const hasMore = firstMessage?.id === lastLoadMessageId;
```
Comment (nguyentrannhan) [inline libs/store/src/lib/messages/messages.slice.ts]: Check ko can firstMessage. Xoa messages.find. Check lastLoadMessage.isFirst

File: libs/store/src/lib/messages/messages.slice.ts [hunk 4]
```diff
@@ -595,16 +595,22 @@ export const messagesSlice = createSlice({
...
-		setNewMessageToUpdateImage(state, action: PayloadAction<ChannelMessage>) {
+		setNewMessageToUpdateImage(state, action) {
 			const data = action.payload;
 			state.newMesssageUpdateImage = {
 				channel_id: data.channel_id,
 				message_id: data.message_id,
 				clan_id: data.clan_id,
 				mode: data.mode,
+				mentions: data.mentions,
+				content: data.content,
```
Comment (nguyentrannhan) [inline libs/store/src/lib/messages/messages.slice.ts:604]: Check neu lay trong onchannelmessage --> compare thoi gian.

File: libs/utils/src/lib/utils/index.ts [hunk 5]
```diff
@@ -646,3 +646,97 @@ export function filterEmptyArrays<T extends Record<string, any>>(payload: T): T
...
+				(attachment): attachment is ApiMessageAttachment => attachment !== null,
+			);
+			setAttachmentPreview(validAttachments);
+		})
+		.catch((error) => {
+			console.error('Error processing files:', error);
+		});
+};
+
+export function base64ToBlob(base64: string, contentType: string, sliceSize = 512): Blob {
```
Comment (nguyentrannhan) [inline libs/utils/src/lib/utils/index.ts]: check show base64 image

File: libs/components/src/lib/components/MessageBox/index.tsx (hunk 1 above)
Comment (phong9xNcc) [reply libs/components/src/lib/components/MessageBox/index.tsx]: fixed: exported to hook

File: libs/components/src/lib/components/MessageBox/FileSelectionButton.tsx (hunk 2 above)
Comment (phong9xNcc) [reply libs/components/src/lib/components/MessageBox/FileSelectionButton.tsx]: fixed: 
image

File: libs/store/src/lib/messages/messages.slice.ts (hunk 4 above)
Comment (phong9xNcc) [reply libs/store/src/lib/messages/messages.slice.ts:604]: image
image
=> newMessageUpdateImage

File: libs/utils/src/lib/utils/index.ts (hunk 5 above)
Comment (phong9xNcc) [reply libs/utils/src/lib/utils/index.ts]: removed that function

File: libs/store/src/lib/messages/messages.slice.ts (hunk 3 above)
Comment (phong9xNcc) [reply libs/store/src/lib/messages/messages.slice.ts]: image
image

**Pull request merged**